src/number_format.js

```diff
@@ -465,7 +467,9 @@ class NumberFormat extends React.Component {
     if (decimalScale !== undefined) afterDecimal = limitToScale(afterDecimal, decimalScale, fixedDecimalScale);
 
     if(thousandSeparator) {
-      beforeDecimal = applyThousandSeparator(beforeDecimal, thousandSeparator, thousandsGroupStyle);
+      const beforeDecimalNumAsString = this.removeFormatting(beforeDecimal);
+      const beforeDecimalFloatValue = parseFloat(beforeDecimalNumAsString);
```

Comment:
I don't think these changes are required. 
The bug is here.
https://github.com/s-yadav/react-number-format/blob/master/src/utils.js#L29
The logic works for cases like `00001,234` but doesn't work only with 0s. We should add check for index being -1. 
Changing
```
const index = str.search(/[1-9]/);
```
to 
```
let index = str.search(/[1-9]/);
index = index === -1 ? str.length : index;
```
on util will fix the bug

Reply:
great feedback.  updated